text/1009-group-by.rst

```diff
@@ -0,0 +1,394 @@
+::
+
+    Status: Draft
+    Type: Feature
+    Created: 2021-12-10
+    Authors: Michael J. Sullivan <sully@msully.net>
+    RFC PR: `edgedb/rfcs#00XX <https://github.com/edgedb/rfcs/pull/XX>`_
+
+==================
+RFC 1009: GROUP BY
+==================
+
+This RFC proposes adding a `GROUP ... BY` statement to EdgeQL to
+perform aggregation.
+
+
+Motivation
+==========
+
+Aggregation of data by keys is a key operation in analytics-like
+queries, and EdgeQL currently serves poorly.
+
+Consider trying to, in our "cards" test basebase representing a
+collectible card game, finding the average cost of a card for each of
+the different "elements" are card could have. In EdgeQL now, this can
+be expressed as::
+
+  FOR element in (DISTINCT Card.element) UNION (
+      WITH g := (SELECT Card FILTER .element = element)
+      SELECT {
+          element := element,
+          avg_cost := sum(g.cost) / count(g)
+      } ORDER BY .element
+  )
+
+This query, while it gets the job done is pretty ugly and
```

Comment:
"pretty ugly" -> "not straightforward"?

"ugly" is very subjective and fwiw it doesn't look ugly to me. Also I'd stick with more formal English in our RFCs.